packages/arb-token-bridge-ui/src/components/UnsupportedSmartContractWalletDisplay/UnsupportedSmartContractWalletDisplay.tsx

```diff
@@ -0,0 +1,121 @@
+import React, { useEffect, useState, useMemo, useCallback } from 'react'
+
+import { BridgeHelper, Inbox__factory } from 'arb-ts'
+import { utils, constants } from 'ethers'
+import { Bridge } from 'token-bridge-sdk'
+
+import { useAppState } from '../../state'
+import { Alert } from '../common/Alert'
+import { Button } from '../common/Button'
+
+const UnsupportedSmartContractWalletDisplay = ({
+  bridge
+}: {
+  bridge: Bridge | null
+}): JSX.Element => {
+  if (!bridge) {
+    return <>...</>
+  }
+  const {
+    app: { networkDetails }
+  } = useAppState()
+  const [recoverableEth, setRecoverableEth] = useState(constants.Zero)
+  const [recoveryAddress, setRecoveryAddress] = useState('')
+
+  useEffect(() => {
+    const checkReverableEth = async () => {
+      const l1ContractAddress = await bridge.l1Bridge.getWalletAddress()
+      const l2Alias = utils.hexValue(
+        BridgeHelper.applyL1ToL2Alias(l1ContractAddress)
+      )
+      const recoverableEthBalance = await bridge.l2Provider.getBalance(l2Alias)
+      setRecoverableEth(recoverableEthBalance)
+    }
+    checkReverableEth()
+  }, [bridge])
+
+  const readableEth = useMemo(() => {
+    return utils.formatEther(recoverableEth)
+  }, [recoverableEth])
+
+  const recover = useCallback(
+    async e => {
+      e.preventDefault()
+      if (networkDetails?.isArbitrum) {
+        return alert('Connect to L1')
+      }
+
+      const inboxAddress =
+        bridge.l1Bridge.network.ethBridge &&
+        bridge.l1Bridge.network.ethBridge.inbox
+      if (!inboxAddress) throw new Error('no inbox address for current network')
+
+      const l1ContractAddress = await bridge.l1Bridge.getWalletAddress()
+      const l2Alias = utils.hexValue(
+        BridgeHelper.applyL1ToL2Alias(l1ContractAddress)
+      )
+      const retryableParams = await bridge.getRetryableTxnParams(
+        '0x',
+        l2Alias,
+        recoveryAddress,
+        recoverableEth,
+        {
+          includeL2Callvalue: false
+        }
+      )
+      const inbox = Inbox__factory.connect(inboxAddress, bridge.l1Signer)
```

Comment:
can't we instead do `bridge.l1Bridge.getInbox()` and get an instantiated inbox contract wrapper?